src/dc_apeerstate.c

```diff
@@ -316,7 +320,7 @@ int dc_apeerstate_init_from_header(dc_apeerstate_t* peerstate, const dc_aheader_
 	peerstate->addr                = dc_strdup(header->addr);
 	peerstate->last_seen           = message_time;
 	peerstate->last_seen_autocrypt = message_time;
-	peerstate->to_save             = DC_SAVE_ALL;
+	peerstate->to_save            |= DC_SAVE_ALL | DC_FORCE_REGOSSIP;
```

Comment:
isn't DC_SAVE_ALL almost implying DC_FORCE_REGOSSIP?  What would happen if we implicitely assume DC_FORCE_REGOSSIP with DC_SAVE_ALL?

Reply:
when the prefer-encrypt or a peer changes, if an encryption is degraded because of missing autocrypt headers, or if a key is verified, DC_SAVE_ALL is needed but DC_FORCE_REGOSSIP is not needed.

maybe it makes sense to force an REGOSSIP also in these cases. not sure, the idea to keep it separated was to save some traffic.